crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -155,10 +172,31 @@ impl AuthorityStorePruner {
             wb.schedule_delete_range(&perpetual_db.objects, &start_range, &end_range)?;
         }
 
+        // When enable_pruning_tombstones is enabled, instead of using range deletes, we need to do a scan of all the keys
+        // for the deleted objects and then do point deletes to delete all the existing keys. This is because to improve read
+        // performance, we set `ignore_range_deletions` on all read options, and using range delete to delete tombstones
+        // may leak object (imagine a tombstone is compacted away, but earlier version is still not). Using point deletes
+        // guarantees that all earlier versions are deleted in the database.
+        if !object_tombstones_to_prune.is_empty() {
+            let mut object_keys_to_delete = vec![];
+            for ObjectKey(object_id, seq_number) in object_tombstones_to_prune {
+                for (object_key, _object_value) in perpetual_db.objects.iter_with_bounds(
+                    Some(ObjectKey(object_id, VersionNumber::MIN)),
+                    Some(ObjectKey(object_id, seq_number.next())),
+                ) {
+                    assert_eq!(object_key.0, object_id);
+                    object_keys_to_delete.push(object_key);
+                }
+            }
+
+            wb.delete_batch(&perpetual_db.objects, object_keys_to_delete)?;
+        }
+
         if !indirect_objects.is_empty() {
-            let ref_count_update = indirect_objects
-                .iter()
-                .map(|(digest, delta)| (digest, delta.to_le_bytes()));
+            let ref_count_update = indirect_objects.iter().map(|(digest, delta)| {
+                debug!("Indirect object {:?} ref count delta {:?}", digest, delta);
```

Comment:
was this log intended to be a permanent addition or was it just for debugging? seems like it might be a bit spammy, trace might be better

Reply:
Good point. Given that indirect object is not used, this may not be very helpful. It was mostly just for me to debug...